### PR TITLE
chore(next): Hackathon bonus tasks

### PR DESCRIPTION
**Pull request by xlisachan**
## This pull request

- [x] Error page - Create a mapping object that maps a backend error from the Cart to a frontend error title and message. Add support for 2 error messages (basic_error, iap_upgrade_contact_support).
- [x] Success page - Add prototype l10n implementation, and support for English, French and 1 additional language of your choosing (German and Spanish (extra bonus))
- [x] Error page - Add prototype l10n implementation, and support for english, french and 1 additional language of your choosing (German and Spanish (extra bonus))

## Checklist

_Put an `x` in the boxes that apply_

- [x] My commit is GPG signed.

### Bonus 1 - Error mapping of iap_upgrade_contact_support
Screenshot 2024-01-24 at 12 19 00 PM

### Bonus 2 - Success page in German
Screenshot 2024-01-24 at 12 19 53 PM

### Bonus 3 - Error page in Spanish
Screenshot 2024-01-24 at 12 20 39 PM

### CODE REVIEW

File: apps/payments/next/app/[offeringId]/error/page.tsx
Comment (xlisachan) [inline apps/payments/next/app/[offeringId]/error/page.tsx:107]: I'd like to revisit using `<article>`. I removed it in this pull request in order to place `payment-panel` where it typically is located, which caused styling issues.

I thought we had considered `Purchase Details` as a part of the main content?

File: apps/payments/next/app/[offeringId]/error/page.tsx
Comment (xlisachan) [inline apps/payments/next/app/[offeringId]/error/page.tsx:102]: This is currently not a heading... I think if this line is optional, we shouldn't consider it as a heading.

File: apps/payments/next/app/[offeringId]/error/page.tsx
Comment (xlisachan) [inline apps/payments/next/app/[offeringId]/error/page.tsx:142]: Aligned with revisiting the a11y issues with the T&P component.

File: apps/payments/next/app/[offeringId]/success/page.tsx
Comment (xlisachan) [inline apps/payments/next/app/[offeringId]/success/page.tsx:59]: Note: I'm having trouble getting the localized date (see screenshots).

**Comment by xlisachan** [comment]
@mozilla/fxa-l10n, this pull request is a part of an exercise and does not require a review or approval. :)

File: apps/payments/next/app/[offeringId]/error/page.tsx
Comment (sardesam) [reply apps/payments/next/app/[offeringId]/error/page.tsx:107]: Works for me! I would love to sit down and get a final decision on this. We halted the TW epic before we could apply all appropriate landmarks and labels.

I have mixed feelings about `payment-panel` as it is both additional and could be deemed required information. I haven't gotten a concrete answer on what is best. In mobile we can hide this information. Since the hackathon was strictly for desktop, I adjusted the page to make it the most accessible within those constraints only. Please feel free to re-add this section in.

File: apps/payments/next/app/[offeringId]/error/page.tsx
Comment (sardesam) [reply apps/payments/next/app/[offeringId]/error/page.tsx:102]: Aligned.

File: apps/payments/next/app/[offeringId]/error/page.tsx
Comment (sardesam) [reply apps/payments/next/app/[offeringId]/error/page.tsx:142]: It's out of scope for this ticket but would definitely like to revisit.

**Review by sardesam — APPROVED** [review]
r+ !

I added a comment as a suggestion for localizing the date. Let me know if that works, otherwise we can pair to figure out a solution or leave a comment in the code in addition to this PR so we know to fix later.

File: apps/payments/next/app/[offeringId]/success/page.tsx
Comment (StaberindeZA) [reply apps/payments/next/app/[offeringId]/success/page.tsx:59]: I don't think any change is required to format the date. The helper functions `getLocalizedDate` and `getLocalizedDateString` already do all the formatting, we just need to make sure that `date` is in unixSeconds, since that's what's currently expected. (Which is how we get it from Stripe typically)

So in this case changing the code to `date = cart.createdAt / 1000` would suffice.

Comment (StaberindeZA) [reply apps/payments/next/app/[offeringId]/success/page.tsx:59]: Oh, just out of interests sake, cause I had to look this up, [Intl.DateTimeFormat](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/DateTimeFormat/DateTimeFormat#timezone) used by `getLocalizedDateString` defaults to the runtimes default time zone.

That's something to be mindful of, since depending on whether this code is run on the server or the client, it would use the servers runtime or the clients runtime, respectively, and the two might not always be the same.

**Comment by bcolsson** [comment]
> @mozilla/fxa-l10n, this pull request is a part of an exercise and does not require a review or approval. :)

@xlisachan - Is this still the case here?

**Comment by xlisachan** [comment]
> > @mozilla/fxa-l10n, this pull request is a part of an exercise and does not require a review or approval. :)
> 
> @xlisachan - Is this still the case here?

Yep, this is still a part of an exercise and won't require a review - I haven't gotten a chance to go through feedback yet. Thanks for checking in!!

**Comment by xlisachan** [comment]
Per @sardesam's suggestion, I will add a line as a reminder to fix the localized date.

**Review by bcolsson — APPROVED** [review]
Approving without review by request. (These strings appear to be pulled into a path that doesn't get extracted.)

**Pull request merged**
